[CS2113-F14-1] Brokeculator

STeng618 opened the pull request:
Brokeculator helps university students manage their expenses

JianJiaT committed:
Merge pull request #99 from JianJiaT/add-javadoc

Add Javadoc comments for expense adding, deleting, listing and summarising

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

JianJiaT committed:
Add intro and quick start

JianJiaT committed:
Swap orders of add and summarise

JianJiaT committed:
Make use of square brackets to denote optional parameters (same as help command)

JianJiaT committed:
Update HelpCommand for summarising

yeozhishen committed:
Merge pull request #101 from JianJiaT/beautify-ug

Update UG format

yeozhishen committed:
Update UG user stories

yeozhishen committed:
Merge pull request #102 from yeozhishen/update-ug

Update UG

STeng618 committed:
Update expense saving

STeng618 committed:
Update event section in UG

STeng618 committed:
Update DG

STeng618 committed:
Merge pull request #103 from STeng618/v2-miscellaneous

V2 miscellaneous

STeng618 committed:
Fix DG bugs

STeng618 committed:
Merge pull request #104 from STeng618/v2-bug-fixing

Fix DG bugs

wenenhoe commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+### Architecture
+
+The UML diagram below shows the main relationships between the classes in the Brokeculator application.
+![img.png](images/architecture.png)
```
Please use the right symbols to annotate the accessors of the attributes and methods of the class in the diagram.

wenenhoe commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+
+The UML diagram below shows the main relationships between the classes in the category feature.
+![category_class.png](images/category_class.png)
+</br>
```
Use of line break can be done with `\` or `<br>`. This is currently being rendered in the html version of the DG.

wenenhoe reviewed (commented):
Generally good job on the DG. I like the inclusion of a table of contents for ease of navigation. Left some comments for areas for improvement, do take a look at them

Daviancold commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+![img.png](images/architecture.png)
+
+### Category feature
+**Implementation** </br>
```
The `</br>` tag is not rendered properly. It shows as text in the webpage. Perhaps consider puttting this tag on the next line?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110613918/7c96fd83-e129-44d0-bfdf-82a77fbd48f6)

Daviancold commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+
+The `Category` class is supplemented by the following classes to interact with the user:
+- `CategoryCommand` - This class is responsible for handling and executing the commands related to categories 
+- `CategoryParser` It is responsible for parsing the user input
```
Formatting inconsistency, there is no dash when describing `CategoryParser`.

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110613918/2916932e-adbd-47e1-a35b-2d69412cb5f5)

Daviancold commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+2. The `Logic` class directs the user input to the `GeneralInputParser` class, which sees the `summarise` keyword
+in the user input and directs it to the `SummariseParser` class
+3. The `SummariseParser` class parses the user input and returns a `SummariseCommand` object or an `InvalidCommand` object 
+depending on whether the user input is valid or not. A `SummariseCommand` object would store relevant information from
```
Applies to all </br> tags below as well.

Daviancold commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| v1.0 | student | see a basic summary of my expenses to see how much i have spent in total | ------------------ |
+| v1.0 | student | view the expenses I have logged | know how much I have spent |
+| v1.0 | paranoid user | save my expenses into a file | backup locally via a file to prevent data loss |
+| v1.0 | student | have the ability to add expenses | ------- |
```
Perhaps you could make the user stories more complete by filling in the 'so that I can' part.

Daviancold commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started] (#setting-up-getting-started)
```
Broken hyperlink here.

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110613918/8602a1cb-05cb-4d3d-8a6d-1474194e8537)

blackmirag3 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+by the `UI` class to be viewed by the user
+6. Executing an `InvalidCommand` object would instead have its error message printed by the `UI` class to be viewed by the user
+
+# Event feature
```
Should the content under "Event feature" fall under "Design & implementation" header? If so I think can use the ### for the header (etc for sub pointer)

blackmirag3 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+is called to set the dashboard object in the `Category` class.
+This is to allow the `Category` class to access the `ExpenseManager` object stored in the `Dashboard` object.
+
+# Summarising expenses
```
Should the content under "Summarising" fall under "Design & implementation" header? If so I think can use the ### for the header (etc for sub pointers)!

blackmirag3 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started] (#setting-up-getting-started)
+* [Design & implementation](#design--implementation)
```
Can consider including summarising expenses and event features too!

blackmirag3 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,165 @@
+This is to allow the `Category` class to access the `ExpenseManager` object stored in the `Dashboard` object.
+
+# Summarising expenses
+**Implementation** </br>
```
Sorry for nitpicking haha, the </br> here too is being rendered as plaintext on HTML version

mihirheda02 committed:
Change Expense date variable to be LocalDate

mihirheda02 committed:
Change EXPECTED.TXT

mihirheda02 committed:
Fix tests relating to date

mihirheda02 committed:
Add date check in try/catch

STeng618 committed:
Change variable name

STeng618 committed:
Merge branch 'master' into 68-use-localdate

STeng618 committed:
Update AddParser.java

mihirheda02 committed:
Add summarise by dates and add tests

mihirheda02 committed:
Merge remote-tracking branch 'origin/68-use-localdate' into 68-use-localdate

yeozhishen committed:
Merge pull request #105 from mihirheda02/68-use-localdate

Use LocalDate for Expense and add summarise by date

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

yeozhishen committed:
Modify AddParser
Modified it to only allow expenses with either 0 or 2 decimal places

yeozhishen committed:
Update UG
Update add and summarise command of the UG to reflect date and expense
decimal change

mihirheda02 committed:
Add to UG

mihirheda02 committed:
Merge pull request #108 from mihirheda02/106-add-to-ug

Add to UG

yeozhishen committed:
Modify Expense class
Edit getStringRepresentation to display existing expenses as 2 decimal
places when getting string representation. In addition made it parse
for 2 decimal places or 0 decimal places for getExpensesFromFile

yeozhishen committed:
Merge branch 'master' into addparser-amount-decimal-places

STeng618 committed:
Delete data/data.txt

STeng618 committed:
Merge pull request #107 from yeozhishen/addparser-amount-decimal-places

Addparser amount decimal places

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

STeng618 committed:
Fix UG line breaks

STeng618 committed:
Fix PDF issues

STeng618 committed:
Fix UG formatting issues

STeng618 committed:
Fix descriptions in UG

yeozhishen committed:
Edit DG
Modified category load sequence diagram (function naming)

yeozhishen committed:
Merge pull request #109 from yeozhishen/category-load-sequence-uml

Edit DG

yeozhishen committed:
Modify DG
- Basic syntax changes, change of </br> to ,br.
- remove broken hyperlinks

yeozhishen committed:
Modify Category UML, Expense
Fixed getExpenseFromFile, in the event that a category does not exist,
changes the category field to null. This is done the preserve the data
integrity of the event class when loading

yeozhishen committed:
Modify Main Loop Error message, GeneralInputParser
Modified GeneralInputParser to handle errors in which only spaces were
provided

yeozhishen committed:
Modify UI
Edited UI such that it returns an empty string and recreates the scanner
upon receiving input that causes the scanner class to throw and
exception. This will prevent the main loop exception from being invoked

yeozhishen committed:
Modify expense, AddParser
Limit expenses loaded from file via getExpenseFromFile in Expense and
ParseInput in addparser to be up to 7 digits, exluding the decimal
places. This will make it harder to encounter floating point overflow/truncations errors

yeozhishen committed:
Modify Tests
Modify Tests in tandem with the category change, also to make the
category clearer as other tests may use the category FOOD

yeozhishen committed:
Fix 127-incorrect-naming-UG

yeozhishen committed:
Add example outputs for Categories in UG

yeozhishen committed:
Remove duplicate summarize parameters in UG

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

yeozhishen committed:
Merge pull request #141 from yeozhishen/v2.1-fixes

V2.1 fixes

mihirheda02 committed:
Add strict date parsing

JianJiaT committed:
Tweak summarise begin index check

JianJiaT committed:
Change summary printing such that it is always 2 dp

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

JianJiaT committed:
Merge branch 'master' into fix-ped-summarise

JianJiaT committed:
Tweak UG for summarise

JianJiaT committed:
Add new error message in enum class for incorrect list command format

JianJiaT committed:
Tweak list parser to standardise format and reject negative input for amount to list

JianJiaT committed:
Tweak expected input format for list

JianJiaT committed:
Tweak error message in ListParser

JianJiaT committed:
Revert "Tweak error message in ListParser"

This reverts commit 29ce4ff26d8e86fe4f8f828dcc044297fd02092b.

JianJiaT committed:
Tweak error messages in ListParser

JianJiaT committed:
Add new error message in enum class for incorrect delete command format

JianJiaT committed:
Tweak DeleteParser to standardise format and reject indices below 1

JianJiaT committed:
Change tests to account for new expected format

yeozhishen committed:
Add ppp and modify category parsing uml
Made change as per DG review

yeozhishen committed:
Add product intro, FAQ in UG
Added faq regarding corrupted data in the UG, Modified the README to
view the product

yeozhishen committed:
Filled up contributions portion of PPP

JianJiaT committed:
Merge pull request #143 from JianJiaT/fix-ped-summarise

Fix summarise based on PE-D feedback

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

JianJiaT committed:
Create PPP

STeng618 committed:
Merge pull request #145 from yeozhishen/ppp-uml-fixes

PPP and uml fix

JianJiaT committed:
Add template and basic content for PPP

STeng618 committed:
Fix event printing

STeng618 committed:
Remove unnecessary file IO messages

STeng618 committed:
Show corrupted entry

STeng618 committed:
Standardise error messages

STeng618 committed:
Skip blank line input

STeng618 committed:
Fix DG bugs

STeng618 committed:
Merge pull request #144 from JianJiaT/fix-ped-delete-list

Fix delete and list based on PE-D feedback

STeng618 committed:
Merge branch 'master' into v2.1-bug-fixing

# Conflicts:
#	src/main/java/brokeculator/parser/DeleteParser.java
#	src/main/java/brokeculator/parser/ListParser.java

STeng618 committed:
Fix AddParser error prompt

STeng618 committed:
Add hyperlinks to commands and examples in UG

STeng618 committed:
Add ppp

STeng618 committed:
Update commands to addexev and delexev

STeng618 committed:
Merge pull request #148 from STeng618/v2.1-bug-fixing

V2.1 bug fixing

yeozhishen committed:
Merge branch 'master' into ppp-uml-fixes

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

JianJiaT committed:
Fix UG for delete and list

JianJiaT committed:
Add PPP

JianJiaT committed:
Merge branch 'master' into create-ppp

JianJiaT committed:
Merge pull request #149 from JianJiaT/create-ppp

Create PPP and fix UG

mihirheda02 committed:
Add comment explaining 'u' over 'y'

mihirheda02 committed:
Merge pull request #142 from mihirheda02/118-strict-date-exception-handling

Add strict date parsing

yeozhishen committed:
Modify DG
Updated some <br> formatting and updated category laod sequence to the
latest version

yeozhishen committed:
Merge pull request #150 from yeozhishen/dg-fixes

Modify DG

yeozhishen committed:
Add newline in UG FAQ

yeozhishen committed:
Merge pull request #151 from yeozhishen/ug-fixes

Add newline in UG FAQ

mihirheda02 committed:
Add to PPP

mihirheda02 committed:
Merge pull request #153 from mihirheda02/152-add-to-ppp

Add to PPP

mihirheda02 committed:
Integrate JLine library for CLI history

STeng618 committed:
Merge pull request #155 from mihirheda02/154-cli-history-feature

Integrate JLine library for CLI history feature

yeozhishen committed:
Remove CommandEnumerator class
Unused class, therefore, removing it to avoid clutter

yeozhishen committed:
Merge pull request #156 from yeozhishen/remove-command-enumerators

Remove CommandEnumerator class

yeozhishen committed:
Prune old and unused code

yeozhishen committed:
Merge pull request #158 from yeozhishen/v2.1-code-pruning

Prune old and unused code

STeng618 committed:
Update documentation

STeng618 committed:
Change boolean function name

STeng618 committed:
Use a method to check if an expense has an owning event

STeng618 committed:
Get a string for printing events

STeng618 committed:
Refactor long methods

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

yeozhishen committed:
Modify category load uml

STeng618 committed:
Merge branch 'master' into v2.1b-miscellaneous

# Conflicts:
#	src/main/java/brokeculator/parser/AddParser.java

yeozhishen committed:
Edit summary in UG

STeng618 committed:
Merge pull request #160 from STeng618/v2.1b-miscellaneous

V2.1b miscellaneous

yeozhishen committed:
Merge pull request #159 from yeozhishen/v2.1-code-pruning

Modify category load uml

yeozhishen committed:
Add javaDoc comments for Category

mihirheda02 committed:
Fix date parsing feedback

yeozhishen committed:
Merge branch 'master' into v2.1-miscellaneous

yeozhishen committed:
Add JavaDoc comments

mihirheda02 committed:
Merge pull request #161 from mihirheda02/157-date-error-feedback

Fix date parsing feedback

STeng618 committed:
Merge branch 'master' into v2.1b-doc

STeng618 committed:
Use general exception

STeng618 committed:
Add general overview of class structures

yeozhishen committed:
Modify DG
Added some non-functional requirements and some manual testing methods

yeozhishen committed:
Add manual testing in DG

yeozhishen committed:
Modify value proposition of DG

STeng618 committed:
Add application logic overview in DG

STeng618 committed:
Merge pull request #162 from yeozhishen/v2.1-miscellaneous

V2.1 miscellaneous

STeng618 committed:
Merge branch 'master' into v2.1b-doc

# Conflicts:
#	docs/DeveloperGuide.md

yeozhishen committed:
Merge pull request #163 from STeng618/v2.1b-doc

V2.1b doc

yeozhishen committed:
Modify category UML

yeozhishen committed:
Merge pull request #165 from yeozhishen/v2.1-miscellaneous

Modify category UML

yeozhishen committed:
Modify category_sequence
Added return line in uml

yeozhishen committed:
Merge pull request #166 from yeozhishen/v2.1-miscellaneous

Modify category_sequence

STeng618 committed:
Change excessively long names

STeng618 committed:
Remove unnecessary code snippet

STeng618 committed:
Add explanations for EventExpenseManager

STeng618 committed:
Fix diagram bugs

mihirheda02 committed:
Fix javadoc format, add documentation, and fix JLine warning issue

mihirheda02 committed:
Edit DG

mihirheda02 committed:
Merge pull request #167 from mihirheda02/164-add-documentation

Add and Edit Documentation

STeng618 committed:
Add explanations for storage

STeng618 committed:
Merge branch 'master' into dg-bugs

# Conflicts:
#	docs/DeveloperGuide.md
#	src/main/java/brokeculator/event/EventExpenseManager.java
#	src/main/java/brokeculator/storage/parsing/FileKeyword.java

STeng618 committed:
Merge pull request #168 from STeng618/dg-bugs

Dg bugs

yeozhishen committed:
Update PPP

yeozhishen committed:
Merge pull request #169 from yeozhishen/v2.1-miscellaneous

Update PPP

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

yeozhishen committed:
Modify category_load_sequence uml
Add extra activation bar

yeozhishen committed:
Merge pull request #171 from yeozhishen/v2.1-miscellaneous

Modify category_load_sequence uml

STeng618 committed:
Add total amount of expenses in event

JianJiaT committed:
Update sequence diagram for summarising expenses

JianJiaT committed:
Update image file of sequence diagram

JianJiaT committed:
Update DG for summarising expenses

mihirheda02 committed:
Add use cases to DG

STeng618 committed:
Update UG

JianJiaT committed:
Merge pull request #173 from JianJiaT/fix-dg-summarise

Fix sequence diagram and UG for summarising expenses

mihirheda02 committed:
Merge pull request #174 from mihirheda02/170-add-use-cases-dg

Add use cases to DG

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

JianJiaT committed:
Tweak ListParser to return InvalidCommand when amount ot list is 0 or lower

JianJiaT committed:
Update HelpCommand for listing and deleting expenses

mihirheda02 committed:
Edit DG table

mihirheda02 committed:
Merge pull request #176 from mihirheda02/175-edit-dg-user-stories-table

Edit DG table

STeng618 committed:
Fix typos

JianJiaT committed:
Remove unused import in ListParser

yeozhishen committed:
Modify category load uml
Shifted logic to the left

yeozhishen committed:
Merge pull request #179 from yeozhishen/v2.1-miscellaneous

Modify category load uml

STeng618 committed:
Fix DG-formatting and add NFRs

STeng618 committed:
Merge branch 'master' into v2.1-miscellaneous

# Conflicts:
#	docs/DeveloperGuide.md

STeng618 committed:
Merge pull request #180 from STeng618/v2.1-miscellaneous

V2.1 miscellaneous

STeng618 committed:
Fix PDF format

STeng618 committed:
Merge pull request #181 from STeng618/master

Fix PDF format

yeozhishen committed:
Modify DG
Edited the table to allow for proper rendering,
added some glossary terms

yeozhishen committed:
Merge pull request #182 from yeozhishen/v2.1-DG-polishing

DG Polishing

yeozhishen committed:
Modify table format in DG

yeozhishen committed:
Merge pull request #183 from yeozhishen/v2.1-DG-polishing

Modify table format in DG

yeozhishen committed:
Modify DG User Stories table format

yeozhishen committed:
Modify DG Table format

yeozhishen committed:
Merge pull request #184 from yeozhishen/v2.1-DG-polishing

V2.1 dg polishing

yeozhishen committed:
Revert DG Table to old format

yeozhishen committed:
Merge pull request #185 from yeozhishen/v2.1-DG-polishing

Revert DG Table to old format

yeozhishen committed:
Add newline before DG Table

yeozhishen committed:
Merge pull request #186 from yeozhishen/v2.1-DG-polishing

Add newline before DG Table

STeng618 committed:
Add PPP

STeng618 committed:
Merge pull request #187 from STeng618/master

Add PPP

STeng618 committed:
Update AboutUs

STeng618 committed:
Merge pull request #188 from STeng618/master

Update AboutUs

yeozhishen committed:
Update UG, PPP

JianJiaT committed:
Merge pull request #178 from JianJiaT/fix-list-amount

Fix list amount

STeng618 committed:
Fix PPP links

STeng618 committed:
Merge branch 'master' of github.com:AY2324S2-CS2113-F14-1/tp

yeozhishen committed:
Merge pull request #189 from yeozhishen/v2.1-PPP

Update UG, PPP

STeng618 committed:
Merge pull request #190 from STeng618/master

Fix PPP links

JianJiaT committed:
Merge branch 'master' of https://github.com/AY2324S2-F14-1/tp

STeng618 committed:
Update steng618.md

JianJiaT committed:
Update PPP

STeng618 committed:
Update steng618.md

yeozhishen committed:
Update yeozhishen.md

yeozhishen committed:
Merge pull request #193 from yeozhishen/v2.1-PPP

Update yeozhishen.md

JianJiaT committed:
Merge pull request #192 from JianJiaT/update-PPP

Update PPP

STeng618 committed:
Update steng618.md

yeozhishen committed:
Modify PPP, DG
Removed redundant line from DG, update PPP with DG extracts

yeozhishen committed:
Merge pull request #194 from yeozhishen/v2.1-PPP

Modify PPP, DG

yeozhishen committed:
Modify ExitParser
Edited it to be case insensitive when checking for user input

yeozhishen committed:
Merge pull request #195 from yeozhishen/v2.1-miscellaneous

Modify ExitParser

yeozhishen committed:
Update yeozhishen.md hyperlinks

yeozhishen committed:
Merge pull request #196 from yeozhishen/v2.1-PPP

Update yeozhishen.md hyperlinks

mihirheda02 committed:
Add more information to PPP

mihirheda02 committed:
Merge pull request #197 from mihirheda02/191-add-to-ppp

Add more information to PPP

STeng618 committed:
Ban file delimiters

STeng618 committed:
Merge branch 'master' into final-fixes

STeng618 committed:
Fix imports

STeng618 committed:
Merge pull request #198 from STeng618/final-fixes

Final fixes

yeozhishen committed:
Code Refactoring

STeng618 committed:
Fix expense index in summary

STeng618 committed:
Merge pull request #199 from yeozhishen/v2.1-final-fixes

Code Refactoring

STeng618 committed:
Merge pull request #200 from STeng618/final-fixes

Fix expense index in summary

STeng618 committed:
Handle malicious sabotage

yeozhishen committed:
Modify logic
Change brokeculator error message

STeng618 committed:
Add explanations for duplicate expenses and events

STeng618 committed:
Handle spaces in input

STeng618 committed:
Fix DG bugs

yeozhishen committed:
Edit Category UML diagrams

yeozhishen committed:
Merge pull request #201 from yeozhishen/v2.1-final-fixes

Modify logic

yeozhishen committed:
Merge branch 'master' into v2.1-final-fixes

STeng618 committed:
Fix DG bugs

STeng618 committed:
Fix DG bugs

STeng618 committed:
Merge branch 'master' of github.com:AY2324S2-CS2113-F14-1/tp

STeng618 committed:
Fix DG bugs

STeng618 committed:
Fix UG bugs

yeozhishen committed:
Modify Event
Add space to line 111

yeozhishen committed:
Modify DG
Change summarize linked image

yeozhishen committed:
Merge branch 'master' into v2.1-final-fixes

yeozhishen committed:
Modify PPP

yeozhishen committed:
Merge pull request #202 from yeozhishen/v2.1-final-fixes

V2.1 final fixes

yeozhishen committed:
Edit PPP Format to fit page limit

yeozhishen committed:
Merge pull request #203 from yeozhishen/v2.1-final-fixes

Edit PPP Format to fit page limit

STeng618 committed:
Edit file delimiters

STeng618 committed:
Add test data

STeng618 committed:
Fix plural noun for multiple expenses

STeng618 committed:
Replace tests in DG

STeng618 committed:
Add warnings about UG

mihirheda02 committed:
Add page breaks to PPP

STeng618 committed:
Update DeveloperGuide.md

mihirheda02 committed:
Merge pull request #205 from mihirheda02/204-add-page-breaks-to-ppp

Add page breaks to PPP

yeozhishen committed:
Modify error handling for adding expense from file

yeozhishen committed:
Merge pull request #206 from yeozhishen/v2.1-final-fixes

Modify error handling for adding expense from file
